components/script/script_thread.rs

```diff
@@ -739,6 +740,14 @@ impl ScriptThread {
             DisableIncrementalGC(runtime.rt());
         }
 
+        let is_baseline_enabled = get_pref("js.baseline.enabled").as_boolean();
+        let is_ion_enabled = get_pref("js.ion.enabled").as_boolean();
+        unsafe {
+            let rt_opts = RuntimeOptionsRef(runtime.rt());
+            is_baseline_enabled.map(|val| (*rt_opts).set_baseline_(val));
+            is_ion_enabled.map(|val| (*rt_opts).set_ion_(val));
```

Comment:
I missed this before:  Could you change this to use `if let` instead of `.map()`?  This is purely a style preference; we find it more readable than `map` in cases where the return value isn't used.

r=mbrubeck with that change